starknet_py/net/client_models.py

```diff
@@ -241,28 +235,22 @@ class TransactionReceipt:
     # pylint: disable=too-many-instance-attributes
 
     transaction_hash: int
+    execution_status: TransactionExecutionStatus
+    finality_status: TransactionFinalityStatus
+    execution_resources: ExecutionResources
+
     events: List[Event] = field(default_factory=list)
     messages_sent: List[L2toL1Message] = field(default_factory=list)
 
-    execution_status: Optional[TransactionExecutionStatus] = None
-    finality_status: Optional[TransactionFinalityStatus] = None
-    # TODO (#1179): remove status field once devnet is updated
-    status: Optional[
-        TransactionStatus
-    ] = None  # replaced by execution and finality status in RPC v0.4.0-rc1
-
     type: Optional[TransactionType] = None
```

Comment:
nit: I believe `type` is now always reqiured for tx receipts, but I guess this is out of scope for this PR. let's remember to re-check fields in time for 0.6.0 update

Reply:
I think it would make sense to add this change now, since changing `tx_receipt` is already in the scope of this PR and we made every other required property non-optional.

But, if this would take much time and stale this PR, I don't mind pushing it back, so it's up to you 😄